src/shared/exec-util.c

```diff
@@ -365,8 +366,47 @@ static int gather_environment_consume(int fd, void *arg) {
         return r;
 }
 
+ExecCommandFlags exec_command_flags_from_strv(char **ex_opts) {
+        ExecCommandFlags flags = 0, ex_flag = 0;
+        char **opt = NULL;
+
+        STRV_FOREACH(opt, ex_opts) {
+                if (!isempty(*opt)) {
+                        ex_flag = exec_command_flags_from_string(*opt);
+                        if (ex_flag != (ExecCommandFlags) -1)
```

Comment:
please propagate the error here. maybe introduce a new _EXEC_COMMAND_FLAGS_INVALID = -1 enum value for that, that you return here, and check explicitly for.